buildSrc/subprojects/versioning/src/main/kotlin/org/gradle/gradlebuild/versioning/DetermineCommitId.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.gradlebuild.versioning
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.process.internal.ExecActionFactory
+import java.io.ByteArrayOutputStream
+import javax.inject.Inject
+
+
+@Suppress("unused")
+open class DetermineCommitId @Inject constructor(
+    objects: ObjectFactory,
+    layout: ProjectLayout,
+    private val execActionFactory: ExecActionFactory
+) : DefaultTask() {
+
+    @get:OutputFile
+    val commitIdFile = objects.fileProperty()
+        .convention(layout.buildDirectory.file("determined-commit-id.txt"))
+
+    @get:Internal
+    val commitId: Provider<String>
```

Comment:
⭕️ How about qualifying this property as `determinedCommitId`? That would help differentiating between this and the `commitId` variable below.